cmd/skopeo/copy.go

```diff
@@ -72,16 +73,36 @@ See skopeo(1) section "IMAGE NAMES" for the expected format
 	flags.StringSliceVar(&opts.additionalTags, "additional-tag", []string{}, "additional tags (supports docker-archive)")
 	flags.BoolVarP(&opts.quiet, "quiet", "q", false, "Suppress output information when copying images")
 	flags.BoolVarP(&opts.all, "all", "a", false, "Copy all images if SOURCE-IMAGE is a list")
+	flags.StringVar(&opts.listProcessing, "list-processing", "CopySystemImage", "How to process image lists. Choose one of CopySystemImage, CopyAllImages, or CopyNoImages")
 	flags.BoolVar(&opts.removeSignatures, "remove-signatures", false, "Do not copy signatures from SOURCE-IMAGE")
 	flags.StringVar(&opts.signByFingerprint, "sign-by", "", "Sign the image using a GPG key with the specified `FINGERPRINT`")
 	flags.StringVar(&opts.digestFile, "digestfile", "", "Write the digest of the pushed image to the specified file")
 	flags.VarP(commonFlag.NewOptionalStringValue(&opts.format), "format", "f", `MANIFEST TYPE (oci, v2s1, or v2s2) to use in the destination (default is manifest type of source, with fallbacks)`)
 	flags.StringSliceVar(&opts.encryptionKeys, "encryption-key", []string{}, "*Experimental* key with the encryption protocol to use needed to encrypt the image (e.g. jwe:/path/to/key.pem)")
 	flags.IntSliceVar(&opts.encryptLayer, "encrypt-layer", []int{}, "*Experimental* the 0-indexed layer indices, with support for negative indexing (e.g. 0 is the first layer, -1 is the last layer)")
 	flags.StringSliceVar(&opts.decryptionKeys, "decryption-key", []string{}, "*Experimental* key needed to decrypt the image")
+	flags.MarkDeprecated("all", "please use --list-processing=CopyAllImages instead")
 	return cmd
 }
 
+// parseListProcessing parses the list processing selection
+// It returns the copy.ImageListSelection to use with image.Copy option
+func parseListProcessing(listProcessing string) (copy.ImageListSelection, error) {
+	switch listProcessing {
+	case "CopySystemImage":
+		return copy.CopySystemImage, nil
+	case "CopyAllImages":
+		return copy.CopyAllImages, nil
+	// There is no CopyNoImages value in copy.ImageListSelection, but because we
+	// don't provide an option to select a set of images to copy, we can use
+	// CopySpecificImages as CopyNoImages.
+	case "CopySpecificImages", "CopyNoImages":
+		return copy.CopySpecificImages, nil
+	default:
+		return copy.CopyAllImages, fmt.Errorf("unknown image list processing option %q. Choose one of the supported options: 'CopySystemImage', 'CopyAllImages', or 'CopyNoImages'", listProcessing)
```

Comment:
(Non-blocking: This might be possible to implement as a map[string]copy.ImageListSelection`; that would also allow constructing the list of supported options here and in the flag definition in a way that doesn’t require updating three places.

But, sure, this works well enough, and `parseManifestFormat` does the same thing.)

Reply:
I'm not sure familiar with cobra/pflags, I couldn't find a simple "Here's how to do a multiple choice", so I copied the manfiest format.

If you know of any doc I'd be happy to try it out :)